pkg/gke/components/Networking.vue

```diff
@@ -0,0 +1,771 @@
+    enableMasterAuthorizedNetwork(neu) {
+      if (!neu) {
+        this.$emit('update:masterAuthorizedNetworkCidrBlocks', []);
+      }
+    },
+
+    subnetwork(neu) {
+      if (neu) {
+        this.$emit('update:createSubnetwork', false);
+      } else {
+        this.$emit('update:createSubnetwork', true);
+      }
+    }
+  },
+
+  methods: {
+    // when credential/region/zone change, fetch dependent resources from gcp
+    loadGCPData() {
+      this.loadingNetworks = true;
+      this.loadingSubnetworks = true;
+      this.getNetworks();
+      this.getSubnetworks();
+      this.getSharedSubnetworks();
+    },
+
+    getNetworks() {
+      getGKENetworks(this.$store, this.cloudCredentialId, this.projectId, { zone: this.zone, region: this.region }).then((res) => {
+        this.networksResponse = res;
+      }).catch((err) => {
+        this.$emit('error', err);
+      });
+    },
+
+    getSubnetworks() {
+      let { region, zone } = this;
+
+      if (!region && !!zone) {
+        region = `${ zone.split('-')[0] }-${ zone.split('-')[1] }`;
+      }
+      getGKESubnetworks(this.$store, this.cloudCredentialId, this.projectId, { region }).then((res) => {
+        this.subnetworksResponse = res;
+      }).catch((err) => {
+        this.$emit('error', err);
+      });
+    },
+
+    getSharedSubnetworks() {
+      return getGKESharedSubnetworks(this.$store, this.cloudCredentialId, this.projectId, { zone: this.zone, region: this.region }).then((res) => {
+        this.sharedSubnetworksResponse = res;
+      }).catch((err) => {
+        this.$emit('error', err);
+      });
+    },
+
+    /**
+     * https://github.com/rancher/rancher/issues/33026 tl;dr:
+     * if networkPolicyEnabled is true, networkPolicyConfig must also be true
+     * if enableNetworkPolicy (project network isolation) is true, networkPolicyEnabled must also be true
+     * networkPolicyConfig does NOT require either enableNetworkPolicy nor networkPolicyEnabled to be true
+     */
+    updateNetworkPolicyEnabled(neu: boolean) {
+      if (neu) {
+        this.$emit('update:networkPolicyEnabled', true);
+        this.$emit('update:networkPolicyConfig', true);
+      } else {
+        this.$emit('update:networkPolicyEnabled', false);
+      }
+    },
+
+    updateEnableNetworkPolicy(neu: boolean) {
+      if (neu) {
+        this.$emit('update:enableNetworkPolicy', true);
+        this.$emit('update:networkPolicyEnabled', true);
+        this.$emit('update:networkPolicyConfig', true);
+      } else {
+        this.$emit('update:enableNetworkPolicy', false);
+      }
+    },
+
+    toggleAdvanced() {
+      this.showAdvanced = !this.showAdvanced;
+    },
+  },
+
+  computed: {
+    ...mapGetters({ t: 'i18n/t' }),
+
+    subnetworks() {
+      return this.subnetworksResponse.items || [];
+    },
+
+    sharedNetworks(): {[key: string]: GKESubnetwork[]} {
+      const allSharedSubnetworks = this.sharedSubnetworksResponse?.subnetworks || [];
+      const networks = {} as any;
```

Comment:
nitpick, --> `const networks: any = {};`

Reply:
did ya one better and typed that badboy correctly